### DISCUSSION TITLE
JSON summary

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#60] [General]
As discussed, we'd like to have some form of a sample-level summary in JSON format. 

Right now, tere are some obstacles to achieving this. Specifically:

- DADA2 versus VSEARCH metrics (this is easily solvable)
- Cutadapt metrics (see below).

For the cutadapt metrics, we currently have two completely different outputs depending on whether we have single-end or paired-end data. Basically, the way cutadapt is meant to be run will not allow us to reliably catch all types of PCR primer site configurations for short amplicons in paired-end reads (5'->3' and 3'-5' - within the same read or one site per read). 

For PE data, we currently get two separate cutadapt JSONS - and right now these are only minimally combined to create a MultiQC custom JSON. The only clean solution would probably be to spend more time on combining these so we get a "complete" Cutadapt JSON from both partial results? I am not sure if this is in fact doable without a fair amount of logic. I tried reading the Multiqc parsing module for cutadapt to see how a minimal format may look like (they probably skip a bunch of stuff..), but it's not exactly easy to get how they do this: https://github.com/MultiQC/MultiQC/blob/main/multiqc/modules/cutadapt/cutadapt.py

Thoughts? Or should we start with a summary JSON without cutadapt for the time being?

**Comment by gregdenay** [comment]
I'm on it. I have the script ready and I'm stilll fiddling around channels for the DADA/VSearch choice (I'm still very slow on Nextflow 🐌).
For the time being I just use the MQC JSON for cutadapt, but it's very bare.

I would tend towards openning an issue to add a script combining cutadapts outputs in a cleaner way/improving on the existing script. This probably doesn't have to be ready for the first release though.
The MultiqcModule class for cutadapt seems to be very robust to any kind of output (except ours), and can handle both logs and jsons. I'm confident we can just adapt the relevant parts to get a mqc compatible JSON with all the info we care about.
